internal/n/reliable_notifier.go

```diff
@@ -0,0 +1,205 @@
+package n
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/capatazlib/go-capataz/internal/s"
+)
+
+// rootName is the name of the ReliableNotifier supervisor root tree node
+var rootName = "reliable-notifier"
+
+// notifierSettings contains settings and callbacks for a ReliableNotifier
+// instance
+type notifierSettings struct {
+	entrypointBufferSize    uint
+	notifierTimeoutDuration time.Duration
+
+	onReliableNotifierFailure func(error)
+	onNotifierTimeout         func(string)
+}
+
+// ReliableNotifierOpt allows clients to tweak the behavior of a
+// ReliableNotifier instance
+type ReliableNotifierOpt func(*notifierSettings)
+
+// newNotifierWorker runs a worker that listens to a channel dedicated to the
+// given event notifier. In the situation the notifierFn panics, the worker gets
+// restarted.
+func newNotifierWorker(
+	_ notifierSettings,
+	name string,
+	notifierFn s.EventNotifier,
+) (chan s.Event, s.Node) {
+	// ch := make(chan s.Event, settings.notifierBufferSize)
+	ch := make(chan s.Event)
+	return ch, s.NewWorker(
+		name,
+		func(ctx context.Context) error {
+			for {
+				select {
+				case <-ctx.Done():
+					return nil
+				case ev := <-ch:
+					notifierFn(ev)
+				}
+			}
+		},
+	)
+}
+
+// newNotifierSubTree runs a subtree of notifier workers. It restarts all
+// notifiers if one of them fails above restart threshold.
```

Comment:
> What is the effect of a notifier worker being restarted? 

All the notifier workers will get restarted at that point, the entrypoint worker will never get restarted given the high tolerance of the root of the tree

> Is it possible to have an infinite failure threshold (or similar) so that a failure in one notifier does not affect others?

You mean, make the `notifiers` sub-tree have infinite tolerance? Yes, I don't see how that would be bad.

>  If a notifier causes the whole notification tree to be restarted, how would you know that has happened? Would you get an event notification somewhere?

Yes, the failure callback would have been triggered in that scenario.

Reply:
> Yes, I don't see how that would be bad.

Ah, I came up with something, we want to know a specific notifier is failing often (observability wise), if I make the `notifiers` sub-tree have an infinite tolerance, then we will never know if you are having a threshold error. 

The other approach would be to have a dedicated supervisor per notifier, but that sounds like a bit too much.

Thoughts @julio-garcia-fc?

Reply:
I think it is necessary to know that a notifier is bouncing. It will always indicate some underlying issue.

Reply:
Agreed, what I'm trying to figure out. Should we: 

a) Keep it as is -- make all notifier restart if one of them keeps failing

b) Only notify errors on the failing notifier and leave others unaffected. The way to achieve this is to have an extra supervisor goroutine per notification function (2 goroutines per notification function)